packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -73,7 +75,8 @@ contract TaikoL1 is EssentialContract, IHeaderSync, V1Events {
         V1Proposing.proposeBlock(state, inputs);
         V1Finalizing.finalizeBlocks(
             state,
-            LibConstants.TAIKO_MAX_FINALIZATIONS_PER_TX
+            LibConstants.TAIKO_MAX_FINALIZATIONS_PER_TX,
+            false
```

Comment:
I changed the code slightly but we still need this `checkHalt` parameter because:

In `proveBlock`, the halt status is checked, but after the check, the status may be  changed from False to True. Therefore, if we call `finalizeBlocks` inside `proveBlock`, we need to skip the checking, otherwise, the function will fail.